tests/unit/CMakeLists.txt

```diff
@@ -240,12 +239,20 @@ add_grb_executables( outer outer.cpp
 	BACKENDS reference reference_omp hyperdags nonblocking
 )
 
+# must generate the golden output for other tests
+force_add_grb_executable( mxv mxv.cpp
+	BACKEND reference
+)
```

Comment:
nice!

Reply:
This is caused by our "un-orthodox" testing modality: the result of `mxv_bsp1d` is compared against the result of `mxv_reference`, which is a "golden output". Therefore, `mxv_reference` must be compiled even when the reference backend was not selected for testing, hence the "force_" prefix (suggestions for a better naming are welcome!).

In other words, dependencies in the testing phase (output of `mxv_bsp1d` checked against output of `mxv_reference`) must be reflected into the compilation phase (compile `mxv_bsp1d` implies compile `mxv_reference`).

Reply:
Yeah I got it, I meant just to compliment the approach=)